src/app/clusters/window-covering-server/window-covering-server.h

```diff
@@ -0,0 +1,224 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/common/gen/attribute-id.h>
+#include <app/common/gen/enums.h>
+#include <app/util/af-types.h>
+#include <map>
+
+class WindowCover
+{
+public:
+    enum class Feature
+    {
+        Lift          = 0x01,
+        Tilt          = 0x02,
+        PositionAware = 0x04
+    };
+
+    struct Mode
+    {
+        uint8_t motorDirReversed : 1; // bit 0
```

Comment:
Is this supposed to line up with the wire encoding?

Standard C++ doesn't specify the layout of bitfields, so if so, the code is not portable.

Most uses of bitfields in this cluster implementation probably need to be converted to bitwise operations on appropriately sized integers.

@rcasallas-silabs

Reply:
@mspang  you mean for endianess ? 
there are translation functions doing the bitwise so this is not an issue